platforms.go

```diff
@@ -121,9 +121,12 @@ import (
 )
 
 var (
-	specifierRe = regexp.MustCompile(`^[A-Za-z0-9_-]+$`)
+	specifierRe    = regexp.MustCompile(`^[A-Za-z0-9_-]+(\([A-Za-z0-9_.-]*\))?$`)
+	osAndVersionRe = regexp.MustCompile(`[()]`)
```

Comment:
I am not sure I understand what the suggestions with the reg ex is - I don't think the reg ex was tight in the first place and the normalize functions are the actual ones that are doing the validation and normalizing part. I would prefer not to enforce any format for the osversion as we do not have an exhaustive list of what the version formatting is for the different OS-es

Reply:
It's not enforcing the OSVersion other than what's already in this PR, but it's to only accept OSVersion in the _first_ element (`os`), and not in any of the other elements (`architecture`, `variant`).

Currently this PR expands the range of accepted characters for _all_ elements, not just the first one, and I think the intent is for only the first one to accept `([A-Za-z ....])`

Reply:
My changes here are only to add the allowance of () in the platform string and keep it aligned mostly to what existed before.

Reply:
Right, so the problem is that before this, all elements (`os`, `arch`, `variant`) accepted the same format (`[A-Za-z0-9_-]+`), which meant that a single regular expression could be used for all elements.

This PR introduces an optional `(OSVersion)`, but (AFAIU) the intent is for that to _only_ apply to the `os` field (so the first element), which means that `arch` and `variant` should no longer share the same regular-expression.

Reply:
![image](https://github.com/containerd/platforms/assets/99994218/ba0cee79-f28b-40a8-a772-bc0f4dc207c4)
this does not pass the existing tests. It will not let windows() or just windows to pass.

Reply:
fixed regex and pushed new changes